tests/main.md

```diff
@@ -329,11 +329,85 @@ val len : int = 5
 - : unit = ()
 ```
 
+Reading with read:
+
+```ocaml
+# let t = Uring.create ~queue_depth:2 ();;
+val t : '_weak6 Uring.t = <abstr>
+
+# let fd = Unix.openfile Test_data.path [ O_RDONLY ] 0;;
+val fd : Unix.file_descr = <abstr>
+# let b1_len = 3 and b2_len = 7;;
+val b1_len : int = 3
+val b2_len : int = 7
+# let b1 = Cstruct.create b1_len and b2 = Cstruct.create b2_len;;
+val b1 : Cstruct.t = {Cstruct.buffer = <abstr>; off = 0; len = 3}
+val b2 : Cstruct.t = {Cstruct.buffer = <abstr>; off = 0; len = 7}
+# Uring.read t fd b1 `Read ~file_offset:Int63.minus_one;;
+- : _[> `Read ] Uring.job option = Some <abstr>
+# Uring.read t fd b2 `Read ~file_offset:Int63.minus_one;;
+- : _[> `Read ] Uring.job option = Some <abstr>
+
+# Uring.submit t;;
+- : int = 2
+
+# let `Read, read = consume t;;
+val read : int = 3
+# Cstruct.to_string b1;;
+- : string = "A t"
+# let `Read, read = consume t;;
+val read : int = 7
+# Cstruct.to_string b2;;
+- : string = "est fil"
+
+# Unix.close fd;;
+- : unit = ()
+```
+
+Writing with write:
+
+```ocaml
+# let t = Uring.create ~queue_depth:2 ();;
+val t : '_weak7 Uring.t = <abstr>
+
+# let rb = Cstruct.create 10 and wb = Cstruct.of_string "Hello";;
+val rb : Cstruct.t = {Cstruct.buffer = <abstr>; off = 0; len = 10}
+val wb : Cstruct.t = {Cstruct.buffer = <abstr>; off = 0; len = 5}
+# let r, w = Unix.pipe ();;
+val r : Unix.file_descr = <abstr>
+val w : Unix.file_descr = <abstr>
+
+# Uring.write t w wb `Write ~file_offset:Int63.minus_one;;
+- : _[> `Write ] Uring.job option = Some <abstr>
+# Uring.read t r rb `Read ~file_offset:Int63.minus_one;;
+- : _[> `Read | `Write ] Uring.job option = Some <abstr>
+
+# Uring.submit t;;
+- : int = 2
+
+# let v, read = consume t;;
```

Comment:
Although likely, I'm not sure the completions are certain to come in order. See e.g.

https://github.com/ocaml-multicore/ocaml-uring/blob/c872cd6a381344eaa30acca65a77b516c1eb68a2/tests/main.md?plain=1#L450-L455